hepcrawl/parsers/arxiv.py

```diff
@@ -0,0 +1,361 @@
+from inspire_utils.helpers import maybe_int, remove_tags
+
+from ..mappings import CONFERENCE_WORDS, THESIS_WORDS
+from ..utils import (
+    coll_cleanforthe,
+    get_first,
+    get_node,
+    split_fullname,
+    ParsedItem,
+    strict_kwargs,
+)
+
+RE_CONFERENCE = re.compile(
+    r'\b(%s)\b' % '|'.join(
+        [re.escape(word) for word in CONFERENCE_WORDS]
+    ),
+    re.I | re.U,
+)
+RE_THESIS = re.compile(
+    r'\b(%s)\b' % '|'.join(
+        [re.escape(word) for word in THESIS_WORDS]
+    ),
+    re.I | re.U,
+)
+
+
+class ArxivParser(object):
+    """Parser for the arXiv format.
+
+    It can be used directly by invoking the :func:`ArxivParser.parse` method, or be
+    subclassed to customize its behavior.
+
+    Args:
+        arxiv_record (Union[str, scrapy.selector.Selector]): the record in arXiv format to parse.
+        source (Optional[str]): if provided, sets the ``source`` everywhere in
+            the record. Otherwise, the source is extracted from the arXiv metadata.
+    """
+    def __init__(self, arxiv_record, source=None):
+        self.root = self.get_root_node(arxiv_record)
+        if not source:
+            source = 'arXiv'
+        self.builder = LiteratureBuilder(source)
+
+    def parse(self):
+        """Extract an arXiv record into an Inspire HEP record.
+
+        Returns:
+            dict: the same record in the Inspire Literature schema.
+        """
+        self.builder.add_abstract(abstract=self.abstract, source=self.source)
+        self.builder.add_title(title=self.title, source=self.source)
+        for license in self.licenses:
+            self.builder.add_license(**license)
+        for author in self.authors:
+            self.builder.add_author(author)
+        self.builder.add_number_of_pages(self.number_of_pages)
+        self.builder.add_publication_info(**self.publication_info)
+        for collab in self.collaborations:
+            self.builder.add_collaboration(collab)
+        for doi in self.dois:
+            self.builder.add_doi(**doi)
+        self.builder.add_preprint_date(self.preprint_date)
+        if self.public_note:
+            self.builder.add_public_note(self.public_note, self.source)
+        for rep_number in self.report_numbers:
+            self.builder.add_report_number(rep_number, self.source)
+        self.builder.add_arxiv_eprint(self.arxiv_eprint, self.arxiv_categories)
+        self.builder.add_private_note(self.private_note)
+        self.builder.add_document_type(self.document_type)
+
+        return self.builder.record
+
+    @property
+    def preprint_date(self):
+        preprint_date_node = self.root.xpath('.//created/text()').extract_first()
+
+        return preprint_date_node
+
+    @property
+    def abstract(self):
+        abstract = self.root.xpath('.//abstract/text()').extract_first()
+
+        return abstract
+
+    @property
+    def authors(self):
+        authors, _, __ = _get_authors_and_collaborations(self.root)
```

Comment:
* `_` can be reused, so you could do `authors, _, _ = ...`
* you're calling several times `_get_authors_and_collaborations`, which might be an expensive function. You can cache it by doing something like:
```python
@property
def authors_and_collaborations(self):
    if not hasatrr(self, '_authors_and_collaborations'):
        self._authors_and_collaborations = _get_authors_and_collaborations(self.root)
    return self._authors_and_collaborations
```
and then using that property everywhere.
Better yet, you can define somewhere a `cached_property` decorator that does this in general and would be used in place of `property` for properties reused several times.